pyiceberg/catalog/rest.py

```diff
@@ -450,6 +450,10 @@ def create_table(
         iceberg_schema = self._convert_schema_if_needed(schema)
         iceberg_schema = assign_fresh_schema_ids(iceberg_schema)
 
+        properties = properties.copy()
```

Comment:
I'm realizing that in the java code base, **write.parquet.compression-codec** is a [table property](https://github.com/apache/iceberg/blob/fb02bd2d713c9bef8b8ccee334f6d65087c344b2/core/src/main/java/org/apache/iceberg/TableProperties.java#L144C30-L144C49), that can be overridden as a [runtime option](https://iceberg.apache.org/docs/latest/spark-configuration/#write-options) on write.

Does it make sense to keep this logic to copy these properties from the catalog properties, or should we simply rely on the users to [set these as table properties](https://github.com/apache/iceberg-python/blob/e385e7757cac08380a7578cf5a230bd62e88bb36/pyiceberg/catalog/rest.py#L448) on table creation?

Reply:
For example, like: 

```
tbl = catalog.create_table(identifier=identifier, schema=TABLE_SCHEMA, properties={'format-version': '1', 'write.parquet.compression-codec': compression_codec, 'write.parquet.compression-level' compression_level})
```